packages/calypso-e2e/src/lib/pages/full-site-editor-page.ts

```diff
@@ -0,0 +1,170 @@
+ */
+export class FullSiteEditorPage {
+	private page: Page;
+
+	private editorToolbarComponent: EditorToolbarComponent;
+	private editorGutenbergComponent: EditorGutenbergComponent;
+	private editorSidebarBlockInserterComponent: EditorSidebarBlockInserterComponent;
+	private editorWelcomeTourComponent: EditorWelcomeTourComponent;
+
+	/**
+	 * Constructs an instance of the page POM class.
+	 *
+	 * @param {Page} page The underlying page.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+
+		this.editorToolbarComponent = new EditorToolbarComponent( page, this.editor );
+		this.editorWelcomeTourComponent = new EditorWelcomeTourComponent( page, this.editor );
+		this.editorSidebarBlockInserterComponent = new EditorSidebarBlockInserterComponent(
+			page,
+			this.editor
+		);
+		// Because of the unique extra iframe in the site editor, this component needs the canvas locator.
+		this.editorGutenbergComponent = new EditorGutenbergComponent( page, this.editorCanvas );
+	}
+
+	/**
+	 * A frame-safe locator to the top level element in the editor, for building other editor locators.
+	 */
+	private get editor(): Locator {
+		return this.page.url().includes( wpAdminPath )
+			? this.page.locator( selectors.editorRoot )
+			: this.page.frameLocator( selectors.editorIframe ).locator( selectors.editorRoot );
+	}
+
+	/**
+	 * A frame-safe locator to the top level element in the editor canvas, for building other editor locators.
+	 * The editor canvas is an iframe wrapping the center editor area that is unique to the site-editor.
+	 */
+	private get editorCanvas(): Locator {
+		return this.editor
+			.frameLocator( selectors.editorCanvasIframe )
+			.locator( selectors.editorCanvasRoot );
+	}
+
+	/**
+	 * Visit the site editor by URL directly.
+	 *
+	 * @param siteHostName Host name of the site, without scheme. (e.g. testsite.wordpress.com)
+	 */
+	async visit( siteHostName: string ): Promise< void > {
+		await this.page.goto( getCalypsoURL( `site-editor/${ siteHostName }` ) );
+	}
+
+	/**
+	 * Waits until the site editor is fully loaded.
+	 */
+	async waitUntilLoaded(): Promise< void > {
+		// There are more stages to the site editor loading than the regular editor.
+		// The most reliable "last" thing to load is the canvas iframe
+		await this.editorCanvas.waitFor( { timeout: 60 * 1000 } );
+		// But then, template parts load async afterwards!
+		const spinnerLocator = this.editorCanvas.locator( selectors.templateLoadingSpinner );
+		// There could be many spinners, so we will keep waiting for the first to be detached.
+		await spinnerLocator.first().waitFor( { state: 'detached' } );
+	}
+
+	/**
+	 * Does all waiting and initial actions to prepare the site editor for interaction.
+	 *
+	 * @param {object} param0 Keyed object of options.
+	 * @param {boolean} param0.leaveWithoutSaving Set if we should auto-except dialog about unsaved changes when leaving.
+	 */
+	async prepareForInteraction(
+		{
+			leaveWithoutSaving,
```

Comment:
Nit: What if we use a destructuring default value:

```
		{
			leaveWithoutSaving = true
		}: {
			leaveWithoutSaving?: boolean;
		}
```

I haven't actually tested, but I think it'd be virtually the same but with less chars.

Reply:
Ah, gotcha! Nah, it's fine the way it is :)